### DISCUSSION TITLE
Save Error

### DISCUSSION THREAD

**Discussion by Twotoolus-FLY-LShst** [voxeltycoon/issues#630] [Troubleshooting] [answered]
**Describe the bug**
Save Error

**Steps to reproduce**
game keep save and fail

**Build version**
0.85.1

**Platform**
win 10 home

**Reproducibility**
?

**Logs/saves**
google link again
https://drive.google.com/file/d/1Ih2AMlguyYCNs5ZiWqN9yfYTl_bdvWB9/view?usp=sharing

![cant save](https://user-images.githubusercontent.com/23209374/115959014-0202da80-a545-11eb-861a-7eee67d366e8.png)

Seed: DRAGON (1983707557)

Saving C:/Users/***/AppData/LocalLow/VoxelTycoon/VoxelTycoon/Saves/fu2n4vc5w3q.sav.tmp...
Metadata                                 1ms                            22.11KB                        0B
SchemaVersionCollection                  47ms                           929B                           51.78MB
AssetLibrary                             0ms                            49.72KB                        0B
TimeManager                              0ms                            12B                            0B
WorldSettings                            0ms                            357B                           0B
CompanyManager                           0ms                            230B                           4KB
VisibilitySettings                       0ms                            12B                            0B
CoverManager                             570ms                          30.92MB                        57.48MB
WorldManager                             836ms                          48.12MB                        0B
GameCameraView                           0ms                            24B                            0B
BuildingManager                          13ms                           461.52KB                       2.73MB
Body                                     2258ms                         39.56MB                        112MB
Overall                                  2260ms                         39.58MB                        112MB

Uploading Crash Report
NullReferenceException: Object reference not set to an instance of an object
  at VoxelTycoon.Tracks.VehicleStation.Write (VoxelTycoon.Serialization.StateBinaryWriter writer) [0x000a3] in <1fb91cae576a4e66bacbd9491a876b8e>:0 
  at VoxelTycoon.Buildings.BuildingManager.Write (VoxelTycoon.Serialization.StateBinaryWriter writer) [0x00081] in <1fb91cae576a4e66bacbd9491a876b8e>:0 
  at VoxelTycoon.Serialization.SaveSerializer.WriteBodyData (System.IO.Stream stream, System.Text.StringBuilder loggerStringBuilder) [0x0004b] in <1fb91cae576a4e66bacbd9491a876b8e>:0 
  at VoxelTycoon.Serialization.SaveSerializer.Write (System.String filename, VoxelTycoon.Serialization.ExtendedSaveMetadata metadata) [0x00096] in <1fb91cae576a4e66bacbd9491a876b8e>:0 
  at VoxelTycoon.Serialization.SaveManager.Write (System.String filename, VoxelTycoon.Serialization.ExtendedSaveMetadata metadata) [0x00013] in <1fb91cae576a4e66bacbd9491a876b8e>:0 
  at VoxelTycoon.Serialization.SaveManager.SaveInternal (VoxelTycoon.Serialization.ExtendedSaveMetadata metadata) [0x0000c] in <1fb91cae576a4e66bacbd9491a876b8e>:0 
  at VoxelTycoon.Serialization.SaveManager.Save (System.String name, System.String filename) [0x0000d] in <1fb91cae576a4e66bacbd9491a876b8e>:0 
  at VoxelTycoon.Game.GameController+<StartSavingInternal>d__18.MoveNext () [0x00075] in <1fb91cae576a4e66bacbd9491a876b8e>:0 
  at UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) [0x00026] in <92f89c20fef94524a39f70be94da1178>:0

**Comment by CrispyBearcon** [comment]
Same problem here,

My save and the 3 autosaves keep "crashing" the game. Game keeps running in background but the "saving..." screen blocks any input for camera movement.


> 
> Uploading Crash Report
> NullReferenceException: Object reference not set to an instance of an object
>   at VoxelTycoon.Tracks.VehicleStation.Write (VoxelTycoon.Serialization.StateBinaryWriter writer) [0x000a3] in <39a46abdc5634327841441811561840c>:0 
>   at VoxelTycoon.Buildings.BuildingManager.Write (VoxelTycoon.Serialization.StateBinaryWriter writer) [0x00081] in <39a46abdc5634327841441811561840c>:0 
>   at VoxelTycoon.Serialization.SaveSerializer.WriteBodyData (System.IO.Stream stream, System.Text.StringBuilder loggerStringBuilder) [0x0004b] in <39a46abdc5634327841441811561840c>:0 
>   at VoxelTycoon.Serialization.SaveSerializer.Write (System.String filename, VoxelTycoon.Serialization.ExtendedSaveMetadata metadata) [0x00096] in <39a46abdc5634327841441811561840c>:0 
>   at VoxelTycoon.Serialization.SaveManager.Write (System.String filename, VoxelTycoon.Serialization.ExtendedSaveMetadata metadata) [0x00013] in <39a46abdc5634327841441811561840c>:0 
>   at VoxelTycoon.Serialization.SaveManager.SaveInternal (VoxelTycoon.Serialization.ExtendedSaveMetadata metadata) [0x0000c] in <39a46abdc5634327841441811561840c>:0 
>   at VoxelTycoon.Serialization.SaveManager.Save (System.String name, System.String filename) [0x0000d] in <39a46abdc5634327841441811561840c>:0 
>   at VoxelTycoon.Game.GameController+<StartSavingInternal>d__18.MoveNext () [0x00075] in <39a46abdc5634327841441811561840c>:0 
>   at UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) [0x00026] in <92f89c20fef94524a39f70be94da1178>:0

**Comment by Airadine** [comment]
I'm having the same problem as well :(

[Player.log](https://github.com/voxeltycoon/bug-tracker/files/6388224/Player.log)

**Comment by Airadine** [comment]
I tried to verify the integrity of the game files. All were successfully validated.

App ID: 732050
Build ID: 6600688

**Comment by andrewpey** [comment]
The problem seems to be caused by **Double Freight Station** mod. If you still experience this problem, please try to disable this mod and let us know if this helps or not.

**Reply by DBT85** [reply]
I have had similar (though I'd have to supply a log to be sure) and don't use that mod at all.

**Reply by andrewpey** [reply]
Thanks for the info, yeah, it would be great to see the log.

**Reply by DBT85** [reply]
[Player.log](https://github.com/voxeltycoon/issues/files/6412213/Player.log)

The log for you. I've not tried the beta branch yet.

**Reply by andrewpey** [reply]
Thanks for the log. Please try to switch to the beta branch and see if it works there. If not, I'll ask you to share the save file with us.

**Comment by andrewpey** [comment, marked as answer]
We've pushed a fix to the [beta branch](https://voxeltycoon.xyz/devlog/beta-branch/), please give it a try.

**Reply by Twotoolus-FLY-LShst** [reply]
GOOD! Finally, can save.
i think, if another user have no problem, close this case.

thank you!
